File: music-application/src/main/java/com/music/sale/application/product/port/inport/ProductQueryUseCase.java
package com.music.sale.application.product.port.inport;

import com.music.sale.application.product.dto.output.GetProductOutput;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface ProductQueryUseCase {
    GetProductOutput getByProductId(Long productId);
    Page<GetProductOutput> getBySellerId(Long sellerId, Pageable pageable);
    Page<GetProductOutput> getByStoreId(Long storeId, Pageable pageable);
    Page<GetProductOutput> searchByKeyword(String keyword, Pageable pageable);
}

File: music-application/src/main/kotlin/com/music/sale/application/product/port/inport/ProductQueryUseCase.kt
package com.music.sale.application.product.port.inport

import com.music.sale.application.product.dto.ProductOutput
import com.music.sale.application.product.dto.SearchProductInput
import org.springframework.data.domain.Page
import org.springframework.data.domain.PageRequest

interface ProductQueryUseCase {
    fun getProducts(pageable: PageRequest): Page<ProductOutput>

    fun getProductById(id: Long): ProductOutput?

    fun searchProducts(
        input: SearchProductInput,
        pageable: PageRequest,
    ): Page<ProductOutput>
} 
